supervisor/process.py

```diff
@@ -209,6 +211,9 @@ def spawn(self):
         self._assertInState(ProcessStates.EXITED, ProcessStates.FATAL,
                             ProcessStates.BACKOFF, ProcessStates.STOPPED)
 
+        if delaystart:
+            time.sleep(self.config.startdelay)
```

Comment:
Supervisor is an asynchronous program.  Blocking like this is unacceptable.

That aside, I'm not sure any kind of delay-before-something features need to be added to Supervisor.  You can have Supervisor run a shell script with some variation of this easily enough:

    #!/bin/sh
    sleep 5
    exec your-program-here